src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -484,6 +484,19 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 
 		let isEdit = options?.params?.isEditConnection ?? false;
 
+		let matcher: interfaces.ProfileMatcher;
+		if (isEdit) {
+			matcher = (a: interfaces.IConnectionProfile, b: interfaces.IConnectionProfile) => a.id === options.params.oldProfileId;
+
+			//Check to make sure the edits are not identical to another connection.
+			await this._connectionStore.isDuplicateEdit(connection, matcher).then(result => {
+				if (result) {
+					this._logService.error(`Profile edit for '${connection.id}' exactly matches an existing profile with data: '${connection.getOptionsKey()}'`);
+					throw new Error(`This profile edit is identical to an already existing profile with data: ${connection.getOptionsKey()}`);
```

Comment:
```suggestion
					throw new Error(`Cannot save profile, the selected connection options are identical to an existing profile with details: \n${connection.getOptionsKey()}`);
```

Reply:
Can you format options Key to a readable form before adding it to error message? The dialog width shows scrollbar if there's no space and looks clumsy.

![image](https://user-images.githubusercontent.com/13396919/226695349-c137bd0b-7a2a-456e-a827-11439bc44952.png)

Would recommend using `; ` for separating keys and `=` for name-value separator for formatted display of optionskey as per connection string semantics.

Reply:
Also include group name if possible as group Id is not useful for end user.

Reply:
> Other than below comments, noticed these issues:
> 
> * [ ]  I'm not able to drag connection with similar properties into server group, whereas can edit connection and modify server group.
> * [ ]  The Query Editor tab titles don't show titles with advanced options (even on hover) - since this is most used feature and users would like to have different tab names based on connection profile differences, I think we should consider displaying full option title on hover on QE tab.
> * [ ]  Similar can be said for Notebook connections and Table Designer connection (on hover) - we should consider showing full options - is it possible to update it universally?
> 
> I'm continuing to test and will update here if I find anything else to be addressed.
> 
> PS. please remove screenshots from description that no longer apply and update with new screenshots (including error dialog)

I changed the title for Query Editor so that it displays all non default options as well as the connection name if set by the user, and since the title is already truncated it shows the basic connection info at the start, with the option to view the rest under the tooltip.